solver/src/main/java/org/chocosolver/solver/constraints/IIntConstraintFactory.java

```diff
@@ -905,9 +905,12 @@ default Constraint allDiffPrec(IntVar[] variables, boolean[][] precedence) {
      * Ensures that all variables from vars take the same value.
      *
      * @param vars list of variables
+     * @implNote
+     * This constraint is reformulated as an atMostNValues constraint, with bound consistency.
      */
     default Constraint allEqual(IntVar... vars) {
-        return atMostNValues(vars, ref().intVar(1), false);
+        return new Constraint(ConstraintsName.ATMOSTNVALUES,
+                new PropAtMostNValues_BC(vars, ref().intVar(1)));
```

Comment:
maybe the PropAtMostNValues_BC should also be available as a atMostNValues constraint ?

Reply:
Well, that's out of the PR, I should create another PR for that so we can discuss it

Reply:
Actually, this was an un-pushed commit done on the master branch...